test/rust-test-util/TestResultReportingClient/src/models/test_result_dto.rs

```diff
@@ -0,0 +1,98 @@
+use chrono::{DateTime, Utc};
+use enumset::EnumSetType;
+use serde::{Deserialize, Serialize};
+
+use crate::MessageTestResult;
+
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+pub(crate) struct TestOperationResultDto {
+    #[serde(rename = "source")]
+    source: String,
+    #[serde(rename = "result")]
+    result: MessageTestResult,
+    #[serde(rename = "type")]
+    test_type: u8,
+    #[serde(rename = "createdAt")]
+    created_at: String,
+}
+
+impl TestOperationResultDto {
+    pub fn new(
+        source: String,
+        result: MessageTestResult,
+        test_type: TestType,
+        created_at: DateTime<Utc>,
+    ) -> Self {
+        Self {
+            source,
+            result,
+            test_type: test_type as u8,
+            created_at: created_at.to_string(),
+        }
+    }
+}
+
+#[derive(Debug, EnumSetType)]
+pub enum TestType {
+    LegacyDirectMethod,
+    LegacyTwin,
+    Messages,
+    DirectMethod,
+    Twin,
+    Network,
+    Deployment,
+    EdgeHubRestartMessage,
+    EdgeHubRestartDirectMethod,
+    Error,
+    TestInfo,
+}
```

Comment:
Even though the client doesn't support all these test types, we need all of them here for the u8 conversion to be correct. I have logic to gate only supported test types in the client.